src/klein/_headers_compat.py

```diff
@@ -0,0 +1,86 @@
+# -*- test-case-name: klein.test.test_headers -*-
+# Copyright (c) 2017-2018. See LICENSE for details.
+
+"""
+Support for interoperability with L{twisted.web.http_headers.Headers}.
+"""
+
+from typing import AnyStr, Iterable, Text, Tuple
+
+from attr import attrib, attrs
+from attr.validators import instance_of
+
+from twisted.web.http_headers import Headers
+
+from zope.interface import implementer
+
+from ._headers import (
+    IMutableHTTPHeaders, RawHeaders, String, headerNameAsBytes,
+    headerValueAsText, normalizeHeaderName, rawHeaderName,
+    rawHeaderNameAndValue,
+)
+
+AnyStr, Iterable, RawHeaders, String, Tuple  # Silence linter
+
+
+__all__ = ()
```

Comment:
* Should be []
* Are we really not exporting any symbols? I feel like anything you expect to be used from outside should be here.

Reply:
Why `[]` and not `()`?

We aren't exporting any of this new API until the all of the (near-term, anyway) pieces are in place, so that we are confident that it has the correct shape before we make any commitments about compatibility.  Also the exports would be done from another file, not from a private `_foo` module.